README.md

```diff
@@ -43,6 +44,32 @@ Thus the above example is equivalent to this:
 var Point = Type({Point: [Number, Number]});
 ```
 
+Instead of supplying only the types of the individual constructors it is also
+possible to define records using object descriptions:
+
+```javascript
+var Point = Type({Point: {x: Number, y: Number}});
+```
+
+Furthermore it is possible to add instance methods. A Maybe type with a map
+function could thus be defined as follows:
+
+```javascript
+var T = function () { return true; };
+var maybeMap(fn) {
+    var this = this;
```

Comment:
Should the variable not be called `self` or something like that rather than `this`?